### ISSUE TITLE
Dynamic allocation with new leads to application crash and error 80000027

### ISSUE THREAD

**Issue by m1awicki** [bebbo/amiga-gcc#330]
Hi. I wanted to port some of my C code to C++. Unfortunately usage of operator `new` leads to runtime crash and error `#80000027`. The very same class allocated on stack works just fine and program executes as expected.
WinUAE screenshot and reproduction sample below. Toolchain is at revision [c8ec483](https://github.com/bebbo/amiga-gcc/commit/c8ec4832ccee21a0d630edef9242670900e22e21).

Compilation command:
` m68k-amigaos-g++ -O3 -fomit-frame-pointer -s -m68000 -mregparm=4 -std=c++17 main.cpp -o da_repro`

```
class Foo
{
};

int main(int argc, char** argv)
{
    Foo* theFoo = new Foo();
    delete theFoo;

    return 0;
}
```

![obraz](https://user-images.githubusercontent.com/41331357/222836038-ec4cf953-f8de-4a5f-ae04-1598c8efd886.png)

**Comment by bebbo** [comment]
please add `-noixemul`

**Comment by m1awicki** [comment]
Thank you for quick reply. Now, compilation command is:
`m68k-amigaos-g++ -O3 -fomit-frame-pointer -s -m68000 -mregparm=4 -std=c++17 -noixemul main.cpp -o da_repro`

It still fails, but more gracefully. Also, attempt to use anything from STL leads to tons of compilation errors related to `regparm`.
E.g.: 
`/opt/amiga/lib/gcc/m68k-amigaos/6.5.0b/include/c++/bits/stl_iterator.h:784:9: error: 'regparm' was not declared in this scope
         __normal_iterator(const __normal_iterator<_Iter,`

WinUAE screenshot:
![obraz](https://user-images.githubusercontent.com/41331357/222896853-7397a246-2596-48ed-a0ff-12308f78118a.png)

**Comment by bebbo** [comment]
The created program here is running without problems. Please attach your binary.

- getting C++ and `-mregparm` to work: no idea if I'll look into this.
- ensure that your stack is large enough.
- use `-fno-exceptions -fno-rtti` if you don't need exceptions

**Comment by m1awicki** [comment]
Thanks for looking into it. Good hint about exceptions and rtti. It is the `-mregparm` that was causing the trouble. I should have figured it out from the error message, but was somehow convinced that this switch should work with g++. Thank you again.

**Comment by bebbo** [comment]
`-mregparm` is working with g++, but the headers aren't prepared for that: there are plenty functions which need the `__stdarg` attribute added.

**Comment by bebbo** [comment]
with the recent fix the correct message is shown:
```
In file included from /opt/amiga/lib/gcc/m68k-amigaos/6.5.0b/include/c++/string:52:0,
                 from main.cpp:1:
/opt/amiga/lib/gcc/m68k-amigaos/6.5.0b/include/c++/bits/basic_string.h: In function 'std::__cxx11::wstring std::__cxx11::to_wstring(double)':
/opt/amiga/lib/gcc/m68k-amigaos/6.5.0b/include/c++/bits/basic_string.h:5636:45: error: invalid conversion from 'int (__attribute__((__stkparm__)) *)(wchar_t*, size_t, const wchar_t*, __gnuc_va_list) {aka int (__attribute__((__stkparm__)) *)(wchar_t*, unsigned int, const wchar_t*, void*)}' to 'int (*)(wchar_t*, std::size_t, const wchar_t*, void*) {aka int (*)(wchar_t*, unsigned int, const wchar_t*, void*)}' [-fpermissive]
     return __gnu_cxx::__to_xstring<wstring>(&std::vswprintf, __n,
```
which points out the type mismatch caused by `-mregparm `
